caom2-compute/src/main/java/ca/nrc/cadc/caom2/compute/CaomWCSValidator.java

```diff
@@ -303,64 +291,121 @@ public static void validatePolarizationWCS(String context, PolarizationWCS polar
-                                    } else {
-                                        axisList[val] = fieldName;
-                                    }
-                                }
-                            }
-                        } catch (IllegalAccessException iae) {
-                            // This error is thrown by the f.get on line 322
-                            throw new IllegalArgumentException(AXES_VALIDATION_ERROR + ": could not access Chunk to validate axes.");
-                        }
-                    }
+            // Go through each axis and validate
+
+            // positionAxis1 and positionAxis2 must both be defined. If they are, position metadata must be defined
+            if (chunk.positionAxis1 != null && chunk.positionAxis1 <= chunk.naxis) {
+                if (chunk.positionAxis2 == null) {
+                    throw new IllegalArgumentException(AXES_VALIDATION_ERROR
+                        + ": positionAxis1 defined (" + chunk.positionAxis1 + ") but positionAxis2 null.");
+
+                }
+                if (chunk.position == null) {
+                    throw new IllegalArgumentException(
+                        String.format(METADATA_NOT_FOUND, AXES_VALIDATION_ERROR, "positionAxis1", chunk.positionAxis1));
+                }
+                if (chunk.positionAxis1 == chunk.positionAxis2) {
+                    throw new IllegalArgumentException(AXES_VALIDATION_ERROR + ": "
+                        + ": duplicate axes: positionAxis1 & positionAxis2 (" + chunk.positionAxis1
+                            + ")");
+                }
+                axisList[chunk.positionAxis1] = "positionAxis1";
+            }
+
+            if (chunk.positionAxis2 != null && chunk.positionAxis2 <= chunk.naxis) {
+                if (chunk.positionAxis1 == null) {
+                    throw new IllegalArgumentException(AXES_VALIDATION_ERROR
+                        + ": positionAxis2 defined (" + chunk.positionAxis2
+                        + ") but positionAxis1 null.");
+                }
+                if (chunk.position == null) {
+                    throw new IllegalArgumentException(
+                        String.format(METADATA_NOT_FOUND, AXES_VALIDATION_ERROR, "positionAxis2", chunk.positionAxis2));
+                }
+                axisList[chunk.positionAxis2] = "positionAxis2";
+            }
+
+            String axisName = "timeAxis";
+            if (chunk.timeAxis != null && chunk.timeAxis <= chunk.naxis) {
+                if (chunk.time == null) {
+                    throw new IllegalArgumentException(
+                        String.format(METADATA_NOT_FOUND, AXES_VALIDATION_ERROR, axisName, chunk.timeAxis)
+                    );
+                } else {
+                    checkDuplicateAxis(axisList, chunk.timeAxis, axisName);
+                }
+                axisList[chunk.timeAxis] = axisName;
+            }
```

Comment:
For the time axis check (but also apply to the other axes so I won't duplicate the comment): 
* the description of timeAxis in the docs 
``index of the time axis; if set: time must have a value```
says nothing about timeAxis <= naxis.

* logically, the checkDuplicateAxis call must always occur before adding the axis to the list so putting it inside the else is unnecessarily more complicated. I know you can't actually skip it but it took more than 0 sec to see that :-)